.github/workflows/cloud-deploy-services.yml

```diff
@@ -0,0 +1,100 @@
+name: Deploy Service
+
+on:
+  workflow_call:
+    inputs:
+      env:
+        description: 'Deployment environment - choose between dev and prod'
+        required: true
+        type: string
+      service:
+        description: 'Service name'
+        required: true
+        type: string
+      package_name:
+        description: 'Docker package name'
+        required: true
+        type: string
+      project_path:
+        description: 'Path to the project directory'
+        required: true
+        type: string
+      test_port:
+        description: 'Service port for testing'
+        required: true
+        type: string
+      local_tag:
+        description: 'Local Docker image tag'
+        required: true
+        type: string
+
+jobs:
+  set_environment:
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ steps.set_env.outputs.environment }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: set_env
+        run: |
```

Comment:
Same applies here. We should eliminate the per environment conditionals as much as possible.